### PR TITLE
nwg-launchers: init at 0.2.0

### PR DESCRIPTION
**Pull request by bbigras**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

`swaylock` needs to be in path. I wonder if we should just wrap `result/bin/nwgbar`.

### CODE REVIEW

File: pkgs/applications/misc/nwg-launchers/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    gtkmm3
+    nlohmann_json
+  ];
+
+  meta = with stdenv.lib; {
+    description = "GTK-based launchers: application grid, button bar, dmenu for sway and other window managers";
+    homepage = "https://github.com/nwg-piotr/nwg-launchers";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+    maintainers = with stdenv.lib.maintainers; [ bbigras ];
```
Comment (IvarWithoutBones) [inline pkgs/applications/misc/nwg-launchers/default.nix:39]:
```suggestion
    maintainers = with maintainers; [ bbigras ];
```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
a minor nit, otherwise LGTM :)

**Comment by bbigras** [comment]
Thanks!

**Comment by IvarWithoutBones** [comment]
commits should be squashed

**Comment by bbigras** [comment]
Done. Yeah. AFAIK there's no way to accept a proposed change and squad it.

Mergers could use "the merge and squash button" but I know the policy is to squash manually.

**Review by IvarWithoutBones — APPROVED** [review]

**Comment by berbiche** [comment]
Relevant comment from the `sgtk-menu` PR https://github.com/NixOS/nixpkgs/pull/93456#issuecomment-660640232

https://github.com/nwg-piotr/nwg-launchers/blob/43f7d5a21715f13f0b5c36529f6f407fc9fbb35a/bar/bar.json#L2-L6

**Comment by bbigras** [comment]
Thanks @berbiche.

I updated the PR to wrap `nwgbar`. I think it might be simpler that way. It will work if someone is using his own bar.json.

The closure's size doesn't seem to change much:
/nix/store/nrhy46w551h9md9hbk69cckxlg76wwcr-nwg-launchers-0.2.0	 182.1M
/nix/store/i5slffzzkv9fw98r4ybgdpq2mkkmmwix-nwg-launchers-0.2.0	 197.0M (with swaylock)

We could have an option to wrap it optionally. And maybe have 2 version in `all-packages.nix` to ensure it's in the cache for everyone but I think it's simpler to just always wrap.

I'm very open to comments. Especially from people with more experience with Nix than me.

**Review by infinisil — APPROVED** [review]
Looking good! I think always wrapping is fine because it doesn't add that much.

**Pull request merged**